webapp/src/main/java/rocks/metaldetector/service/notification/NotificationServiceImpl.java

```diff
@@ -35,27 +39,18 @@
 
   @Override
 //  @Scheduled(cron = "0 0 4 * * SUN")
-  public void notifyAllUsers() {
-    userService.getAllActiveUsers().forEach(user -> notifyUser(user.getPublicId()));
-  }
+  @Transactional
+  public void notifyOnFrequency() {
+    List<ReleaseContainer> releaseContainer = createReleaseContainer();
 
-  @Override
-  public void notifyUser(String publicUserId) {
-    UserDto user = userService.getUserByPublicId(publicUserId);
-    List<String> followedArtistsNames = followArtistService.getFollowedArtistsOfUser(publicUserId).stream()
-        .map(ArtistDto::getArtistName).collect(Collectors.toList());
-
-    if (!followedArtistsNames.isEmpty()) {
-      var now = LocalDate.now();
-      List<ReleaseDto> newReleases = releaseService.findAllReleases(followedArtistsNames, new TimeRange(now, now.plusMonths(3)));
-
-      if (!newReleases.isEmpty()) {
-        emailService.sendEmail(new NewReleasesEmail(user.getEmail(), user.getUsername(), newReleases));
-      }
-    }
+    notificationConfigRepository.findAll().stream()
+        .filter(config -> config.getUser().isEnabled() &&
+                          config.getNotify())
+        .forEach(notificationConfig -> frequencyNotification(notificationConfig, getReleaseContainerForConfig(releaseContainer, notificationConfig)));
```

Comment:
I'm confused aboud `config` vs. `notificationConfig` (see `filter` lambda above)

```suggestion
        .forEach(config -> frequencyNotification(config, getReleaseContainerForConfig(releaseContainer, config)));
```